File: lint-arwaky/archive/contract/output_client_aggregate.py
from abc import ABC, abstractmethod
from typing import Iterator
import io
from ..taxonomy import FilePath, LogOutput, Identity, FileFormat

class OutputClientAggregate(ABC):
    """AGGREGATE: Orchestrator for managing output files and stdout teeing."""
    @abstractmethod
    def get_output_dir(self) -> FilePath | None:
        ...

    @abstractmethod
    def write_output(
        self,
        output: LogOutput,
        command: Identity,
        output_format: FileFormat | None = None,
    ) -> FilePath | None:
        ...

    @abstractmethod
    def tee_stdout(self) -> Iterator[io.StringIO]:
        ...

File: lint-arwaky/src-rust/contract/output_client_aggregate.rs
use crate::taxonomy::{FilePath, FileFormat};

pub trait OutputClientAggregate: Send + Sync {
    fn get_output_dir(&self) -> Option<&FilePath>;
    fn write_output(&self, output: &str, command: &str, output_format: Option<&FileFormat>) -> Option<FilePath>;
}
